adapters/criteo/criteo.go

```diff
@@ -0,0 +1,101 @@
+package criteo
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v14/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	URI             string
+	slotIDGenerator slotIDGenerator
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, extraRequestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	criteoRequest, errs := newCriteoRequest(a.slotIDGenerator, request)
+	if len(errs) > 0 {
+		return nil, errs
+	}
+
+	jsonRequest, err := json.Marshal(criteoRequest)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	rqData := adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.URI,
+		Body:    jsonRequest,
+		Headers: getCriteoRequestHeaders(&criteoRequest),
+	}
+
+	return []*adapters.RequestData{&rqData}, nil
+}
+
+func getCriteoRequestHeaders(criteoRequest *criteoRequest) http.Header {
+	headers := http.Header{}
+
+	if criteoRequest != nil {
+		if criteoRequest.User.CookieID != "" {
+			headers.Add("Cookie", "uid="+criteoRequest.User.CookieID)
+		}
+
+		if criteoRequest.User.IP != "" {
+			headers.Add("X-Client-Ip", criteoRequest.User.IP)
+		}
+
+		if criteoRequest.User.UA != "" {
+			headers.Add("User-Agent", criteoRequest.User.UA)
+		}
+	}
+
+	return headers
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	// TODO = Handle more bad response.StatusCode
```

Comment:
Please resolve the TODO comments. This was fine when the adapter was still in development, but as we get closer to merging they should be handled.

Reply:
No worries.